script.js

```diff
@@ -0,0 +1,174 @@
+const key = `82005d27a116c2880c8f0fcb866998a0`;
+const inputData = document.getElementById("input-display");
+const search = document.getElementById("search");
+const weatherImg = document.querySelector(".img-fluid");
+const weatherTemp = document.querySelector(".temp-details");
+const weatherCondition = document.querySelector(".weather-conditions");
+const cityLocation = document.querySelector(".city");
+const countryLocation = document.querySelector(".country");
+const loader = document.querySelector(".loader");
+const container = document.querySelector(".weather-container");
+const disable = document.querySelector(".disable");
+const errorImg = document.querySelector(".img-fluiddd");
+const errorMessage = document.querySelector(".weather-content");
+const cards = document.querySelector(".cards");
+const dayNames = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+let flag = true;
+const date = new Date();
+let cardContent = '';
+let spinner = false;
+const getDataForDay = (weatherItem) => {
+  const day = dayNames[new Date(weatherItem.dt_txt).getDay()];
+  const temp = weatherItem.main.temp;
+  const tempInCelsius = Math.floor(temp - 273.15) + "°"
+  const description = weatherItem.weather[0].description;
+  const iconsCode = weatherItem.weather[0].icon;
+  const weatherStatus = weatherItem.weather[0].main;
+  return { tempInCelsius, description, iconsCode, weatherStatus, day };
+}
+
+const currentLocationWeather = () => {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(async function (position) {
+      const latitude = position.coords.latitude;
+      const longitude = position.coords.longitude;
+      const weatherDeatils = await fetchData(latitude, longitude);
+      weatherDeatils.forEach(element => {
+        displayWeatherInfo(element);
+      })
+
+    }, () => {
+      disable.style.display = "block";
+      container.style.display = "none";
+      cards.style.display = "none";
+      errorImg.src = `icons/unknown.png`;
+      errorMessage.innerHTML = "PERMISSION BLOCKED";
+    });
+  }
+}
+
+currentLocationWeather();
+const handleInputData = async () => {
+  cardContent = "";
+  flag = true;
+  if (inputData.value.trim() == "") {
+    alert("Enter city name");
+    inputData.value = "";
+  }
+  else {
+    const res = await getLatitudeLongitude(inputData.value);
+    const lat = res?.lat
+    const lon = res?.lon
+    if(lat && lon){
+      const weatherDeatils = await fetchData(lat, lon);
+      inputData.value = "";
+      weatherDeatils.forEach(element => {
+        displayWeatherInfo(element);
+      })}
+    else{
+      disable.style.display = "block";
+      container.style.display = "none";
+      cards.style.display = "none";
+      errorImg.src = `icons/unknown.png`;
```

Comment:
don't repeat code create a separate function for this and call whenever you need